### DISCUSSION TITLE
(every? string? '()) => true

### DISCUSSION THREAD

**Discussion by wandersoncferreira** [clj-br/forum#25] [Geral]
Vira e mexe eu caio no problema da "Verdade por vacuidade" https://pt.wikipedia.org/wiki/Verdade_por_vacuidade essa entrada do clojure docs é bem importante https://clojuredocs.org/clojure.core/every_q#example-542692d3c026201cdc326fa1 =)

**Comment by lsevero** [comment]
Basicamente toda verificação com `every? ` precisa ter um `(and (not (empty? ...)) ...)` junto pra evitar falsos positivos. Se a coleção for vazia o every? não faz sentido.

**Reply by ericdallo** [reply]
Acho que dá pra trocar esse and not empty por `seq`

**Reply by wandersoncferreira** [reply]
> Acho que dá pra trocar esse and not empty por `seq`

engraçado que quase todos os times que trabalhei até hoje preferem usar not empty do que seq apesar de ser a forma recomendada de fazer essa validação.

**Reply by ericdallo** [reply]
Sim, realmente existe esse tipo de discussao 😅

**Reply by mauricioszabo** [reply]
Eu sempre preferi `not-empty` (a função) ao invés de `seq` porque achava que fazia só uma verificação mesmo.

Até descobrir que not-empty faz **mais coisas** do que seq (inclusive é implementado usando seq hahahaha)
